pkg/apis/core/validation/validation.go

```diff
@@ -6194,8 +6195,21 @@ func validateServiceTrafficDistribution(service *core.Service) field.ErrorList {
 		return allErrs
 	}
 
-	if *service.Spec.TrafficDistribution != v1.ServiceTrafficDistributionPreferClose {
-		allErrs = append(allErrs, field.NotSupported(field.NewPath("spec").Child("trafficDistribution"), *service.Spec.TrafficDistribution, []string{v1.ServiceTrafficDistributionPreferClose}))
+	var supportedTrafficDistribution []string
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PreferSameTrafficDistribution) {
```

Comment:
pedantically, this could be something like:

```
var supportedTrafficDistribution = func() sets.Set[string] {
    if !utilfeature.DefaultFeatureGate.Enabled(features.PreferSameTrafficDistribution) {
	    return sets.New(v1.ServiceTrafficDistributionPreferClose)
    }
	return sets.New(
			v1.ServiceTrafficDistributionPreferClose,
			v1.ServiceTrafficDistributionPreferSameZone,
			v1.ServiceTrafficDistributionPreferSameNode)
}()

func validateServiceTrafficDistribution(...) {
  ...
}
```

But service create/update isn't nearly as frequent as endpoints, so it matters a little less.

Reply:
That wouldn't work though, because `--feature-gates` hasn't been processed yet at variable-initialization time, right? (And also it would break `SetFeatureGateDuringTest`.)

Reply:
oh, right, good point.  Would need an accessor function and a sync.Once.  Nevermind.